### PR TITLE
feat(scripts): add script to get and write a CSV from the issues in this repository

### PR DESCRIPTION
**Pull request by jcao02**

### CODE REVIEW

File: scripts/export_csv.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+          }
+        }
+      }
+    }
+  """ })
+  response = requests.post(GITHUB_ENDPOINT, data=payload, headers=HEADERS).json()
+  hum_response = helpers.humanize_graphql_response(response['data'])
+  return hum_response['organization']['repository']['issues']['totalCount']
+
+# Performs a request to get the total number of issues
```
Comment (chamini2) [inline scripts/export_csv.py]: ~issues~ -> labels

File: scripts/export_csv.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  hum_response = helpers.humanize_graphql_response(response['data'])
+  return hum_response['organization']['repository']['labels']['totalCount']
+
+# Gets the issues from the server 
+def get_issues():
+  total_issues = get_issues_count()
+  print "Total issues: %s" % total_issues
+
+  total_labels = get_labels_count()
+
```
Comment (chamini2) [inline scripts/export_csv.py]: Remove this line

File: scripts/export_csv.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+            edges {
+              node {
+                number,
+                title,
+                createdAt,
+                closedAt,
+                author {
+                  login
+                },
+
```
Comment (chamini2) [inline scripts/export_csv.py]: remove line, right?

File: scripts/export_csv.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  payload = build_payload(total_issues, total_labels)
+
+  response = requests.post(GITHUB_ENDPOINT, data=payload, headers=HEADERS).json()
+
+  parsed_response = helpers.humanize_graphql_response(response['data'])
+  return parsed_response['organization']['repository']['issues']
+
+# Builds a payload to get totalCount issues from the server
+def build_payload(total_issues, total_labels):
+  issues_to_get = total_issues if total_issues < ISSUE_LIMIT else ISSUE_LIMIT
```
Comment (chamini2) [inline scripts/export_csv.py]: This may be a problem eventually, but I don't see it in the near future.

File: scripts/export_csv.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  response = requests.post(GITHUB_ENDPOINT, data=payload, headers=HEADERS).json()
+  hum_response = helpers.humanize_graphql_response(response['data'])
+  return hum_response['organization']['repository']['issues']['totalCount']
+
+# Performs a request to get the total number of issues
+def get_labels_count():
+  payload = json.dumps({ 'query': """
+    query {
+      organization(login: "zomoz") {
+        name, url,
```
Comment (chamini2) [inline scripts/export_csv.py]: In different lines

Comment (chamini2) [reply scripts/export_csv.py]: Sometimes you use a comma, sometimes you don't try to be consistent

File: scripts/export_csv.py [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+              node {
+                number,
+                title,
+                createdAt,
+                closedAt,
+                author {
+                  login
+                },
+        
+                assignees(last: 1) {
```
Comment (chamini2) [inline scripts/export_csv.py]: Are we sure we just want one? Always?

File: scripts/export_csv.py [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+  for issue in issues:
+    csv_out.writerow((
+      issue['number'],
+      helpers.format_date(issue['createdAt']),
+      helpers.format_date(issue['closedAt']),
+      issue['impact'],
+      issue['weight'],
+      1 if issue['blocking'] else 0,
+      ','.join(issue['modules']),
```
Comment (chamini2) [inline scripts/export_csv.py]: Will this add more columns to the CSV or is it a single column with commas encolsed in quotes?

File: scripts/export_csv.py [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+    ))
+
+  print "%s written" % csv_file
+
+
+## MAIN FUNCTION
+def main():
+  issues = get_issues()
+  parsed_issues = parse_issues(issues)
+  write_csv(parsed_issues)
```
Comment (chamini2) [inline scripts/export_csv.py]: You could also write a CSV header with the column names

File: scripts/export_csv.py (hunk 7 above)
Comment (jcao02) [reply scripts/export_csv.py]: Single column with comma-separated values

File: scripts/export_csv.py (hunk 8 above)
Comment (jcao02) [reply scripts/export_csv.py]: It's done inside the function `write_csv`

File: scripts/export_csv.py (hunk 6 above)
Comment (jcao02) [reply scripts/export_csv.py]: Yes! An issue will only have a single responsible for our metrics

File: scripts/export_csv.py [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+# in a GitHub repository
+
+import requests
+import json
+import helpers
+import datetime
+import csv
+
+# Generate this using: https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/
+PERSONAL_ACCESS_TOKEN = ''
```
Comment (zargex) [inline scripts/export_csv.py]: We have a zomoz-bot user, maybe it's a good idea using its token

Comment (chamini2) [reply scripts/export_csv.py]: Agreed!

Comment (jcao02) [reply scripts/export_csv.py]: We should not commit the token anyway, should we?

Comment (chamini2) [reply scripts/export_csv.py]: Well, zomoz-bot only has reading permissions I think, but you are probably right.

Comment (chamini2) [reply scripts/export_csv.py]: Specially since this is a public repo for now.

Comment (zargex) [reply scripts/export_csv.py]: @chamini2 zomoz-bot has writing permissions, so jenkins can mark the commit status.

**Pull request merged**